### PR TITLE
`azurerm_kubernetes_cluster` - making `linux_profile` optional

### PR DESCRIPTION
**Pull request by metacpp**
This PR is going to resolve the #1806.

The latest AKS API does not require linux_profile to be set, and Azure portal follows it. Terraform still requires it by default, this inconsistency caused the importing issue.

Manually created instance on Azure portal, and did `terraform import` and `terraform plan`, no change is required.

### CODE REVIEW

**Comment by metacpp** [comment]
@zqingqing1 feel free to review it.

**Comment by zqingqing1** [comment]
lgtm

**Comment by metacpp** [comment]
Test automation result:
screen shot 2018-08-24 at 8 25 01 pm

File: azurerm/resource_arm_kubernetes_cluster_test.go [hunk 1]
```diff
@@ -357,7 +349,7 @@ resource "azurerm_kubernetes_cluster" "test" {
     client_secret = "%s"
   }
 }
-`, rInt, location, rInt, rInt, rInt, clientId, clientSecret)
+`, rInt, location, rInt, rInt, clientId, clientSecret)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_kubernetes_cluster_test.go]: can you add a separate test case (and import test) for this which includes the linux profile block e.g. `TestAccAzureRMKubernetesCluster_linuxProfile`

**Pull request merged**